packages/next/build/webpack-config.ts

```diff
@@ -19,134 +19,124 @@ import {ServerlessPlugin} from './webpack/plugins/serverless-plugin'
 // The externals config makes sure that
 // on the server side when modules are
 // in node_modules they don't get compiled by webpack
-function externalsConfig (isServer, target) {
-  const externals = []
-
+function externalsConfig (isServer, target): webpack.ExternalsElement | webpack.ExternalsElement[] | undefined {
   // When the serverless target is used all node_modules will be compiled into the output bundles
   // So that the serverless bundles have 0 runtime dependencies
   if (!isServer || target === 'serverless') {
-    return externals
+    return
   }
 
   const notExternalModules = ['next/app', 'next/document', 'next/link', 'next/router', 'next/error', 'http-status', 'string-hash', 'ansi-html', 'hoist-non-react-statics', 'htmlescape']
 
-  externals.push((context, request, callback) => {
+  return (context, request, callback) => {
```

Comment:
Looks like this is a breaking change 🤔 you're returning a function instead of an array. Same with returning undefined. Users could be assuming it's an array currently.

Reply:
That's the value of the config. `Externals | Externals[] | undefined`. I'm going to refactor a lot more to fix the remaining errors.

Reply:
What I mean is that the `externals` key should be an array otherwise it's a breaking change.

Reply:
An empty array is invalid webpack config

Reply:
sounds like that's wrong in the types